opentelemetry-sdk/src/opentelemetry/sdk/logs/__init__.py

```diff
@@ -112,6 +115,133 @@ def force_flush(self, timeout_millis: int = 30000):
         """
 
 
+class SynchronousMultiLogProcessor(LogProcessor):
+    """Implementation of class:`LogProcessor` that forwards all received
+    events to a list of log processors sequentially.
+
+    The underlying log processors are called in sequential order as they were
+    added.
+    """
+
+    def __init__(self):
+        # use a tuple to avoid race conditions when adding a new log and
```

Comment:
Is the tuple what avoids race conditions or is it the lock in `add_log_processor`? A tuple is an ordered data structure, is the order meaningful for log processors? If not, I suggest this be a `set`, instead.

Reply:
It's both tuple and the lock. We are trying to be consistent with span processors so the order is important here.